### PR TITLE
e2e: refactor and improve lb test

### PR DESCRIPTION
**Pull request by zirain**
https://github.com/envoyproxy/gateway/actions/runs/10018597540/job/27694209720

### CODE REVIEW

File: test/e2e/tests/load_balancing.go [hunk 1]
```diff
@@ -80,40 +81,62 @@ var RoundRobinLoadBalancingTest = suite.ConformanceTest{
...
+			if err := wait.PollUntilContextTimeout(context.TODO(), time.Second, 30*time.Second, true, func(_ context.Context) (bool, error) {
+				return runTrafficTest(t, suite, req, expectedResponse, sendRequests, compareFunc), nil
+			}); err != nil {
+				tlog.Errorf(t, "failed to run round robin load balancing test: %v", err)
 			}
 		})
 	},
 }
 
+type TrafficCompareFunc func(trafficMpa map[string]int) bool
```
Comment (shawnh2) [inline test/e2e/tests/load_balancing.go:104]:
```suggestion
type TrafficCompareFunc func(trafficMap map[string]int) bool
```

**Review by shawnh2 — COMMENTED** [review]
LGTM thanks for optimizing it

**Review by shawnh2 — APPROVED** [review]

**Comment by zirain** [comment]
wait https://github.com/envoyproxy/gateway/pull/3913

**Review by shawnh2 — APPROVED** [review]

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**